From 89d25a1c027688400371bcc9b4eb4d1481edee00 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 28 Apr 2023 11:52:24 -0400
Subject: [PATCH 21/35] Add a make rule for compile_commands.json

This adds a make rule to generate compile_commands.json, which some
verifier tools depend on.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 .gitignore | 1 +
 Makefile   | 6 +++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index 7fc55bc..594090e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -33,6 +33,7 @@ Make.local
 /.cache/
 /certdb/
 /compile_commands.json
+/compile_commands.events.json
 /cov-int/
 /post-process-pe
 /random.bin
diff --git a/Makefile b/Makefile
index f0f53f8..61077bd 100644
--- a/Makefile
+++ b/Makefile
@@ -76,6 +76,10 @@ ifneq ($(origin EFI_PATH),undefined)
 	$(error EFI_PATH is no longer supported, you must build using the supplied copy of gnu-efi)
 endif
 
+compile_commands.json : Makefile Make.rules Make.defaults 
+	make clean
+	bear -- make COMPILER=clang test all
+
 update :
 	git submodule update --init --recursive
 
@@ -322,7 +326,7 @@ clean-lib-objs:
 
 clean-shim-objs:
 	@rm -rvf $(TARGET) *.o $(SHIM_OBJS) $(MOK_OBJS) $(FALLBACK_OBJS) $(KEYS) certdb $(BOOTCSVNAME)
-	@rm -vf *.debug *.so *.efi *.efi.* *.tar.* version.c buildid post-process-pe
+	@rm -vf *.debug *.so *.efi *.efi.* *.tar.* version.c buildid post-process-pe compile_commands.json
 	@rm -vf Cryptlib/*.[oa] Cryptlib/*/*.[oa]
 	@if [ -d .git ] ; then git clean -f -d -e 'Cryptlib/OpenSSL/*'; fi
 
-- 
2.39.0

